### PR TITLE
Add support for new getTagKeys/getTagValues interface

### PR DESCRIPTION
**Pull request by kaydelaney**
Sets variable error state if getTagKeys/getTagValues returns an error.
Lots of `@ts-expect-error` since `11.1.x` isn't out yet 😬 

<details>
  <summary>📦 Published PR as canary version: <code>5.1.0--canary.790.9497828604.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/scenes-react@5.1.0--canary.790.9497828604.0
  npm install @grafana/scenes@5.1.0--canary.790.9497828604.0
  # or 
  yarn add @grafana/scenes-react@5.1.0--canary.790.9497828604.0
  yarn add @grafana/scenes@5.1.0--canary.790.9497828604.0
  ```
</details>

### CODE REVIEW

File: packages/scenes/src/variables/groupby/GroupByVariable.tsx [hunk 1]
```diff
@@ -54,7 +55,7 @@ export interface GroupByVariableState extends MultiValueVariableState {
 export type getTagKeysProvider = (
   set: GroupByVariable,
   currentKey: string | null
-) => Promise<{ replace?: boolean; values: MetricFindValue[] }>;
+) => Promise<{ replace?: boolean; values: MetricFindValue[] | GetTagResponse }>;
 
 export class GroupByVariable extends MultiValueVariable<GroupByVariableState> {
```
Comment (dprokop) [inline packages/scenes/src/variables/groupby/GroupByVariable.tsx:60]: @kaydelaney -  while it works for fiters:

image


it doesn't for group by:

image

Comment (kaydelaney) [reply packages/scenes/src/variables/groupby/GroupByVariable.tsx:60]: Thanks for catching, will look into it

Comment (kaydelaney) [reply packages/scenes/src/variables/groupby/GroupByVariable.tsx:60]: Should work now!

Comment (dprokop) [reply packages/scenes/src/variables/groupby/GroupByVariable.tsx:60]: @kaydelaney  - indeed, it does. But now the same problem appears for filter value (gatTagValues call). This is an interesting case as in filters the errors can come both from the dimensions and values calls, and the current UI does not support that. Think showing the last error that has occured should work fine, don't think we want to provide two errors independently for dimensions and for values. WDYT?

Comment (kaydelaney) [reply packages/scenes/src/variables/groupby/GroupByVariable.tsx:60]: That sounds reasonable. Will implement asap

Comment (kaydelaney) [reply packages/scenes/src/variables/groupby/GroupByVariable.tsx:60]: Should be fixed now

**Review by dprokop — APPROVED** [review]
@kaydelaney - looks good now :) Can you please add tests for the error renedering?

**Comment by kaydelaney** [comment]
Gonna merge now as issues with linking scenes/grafana is making writing tests hard atm. :( Will make a follow-up PR when that issue is fixed.

**Pull request merged**